modules/@apostrophecms/asset/index.js

```diff
@@ -37,12 +47,48 @@ module.exports = {
           await fs.mkdirp(buildDir);
           await fs.remove(modulesDir);
           await fs.mkdirp(modulesDir);
-          await fs.remove(bundleDir);
-          await fs.mkdirp(bundleDir);
+
+          let rebuildAposUi = false;
+          const APOS_MERGED_BUNDLE = 'apos-bundle.js';
+          const APOS_ONLY_BUNDLE = 'apos-only-bundle.js';
+          const APOS_ONLY_TS = '.apos-only-timestamp.txt';
+          const PUBLIC_BUNDLE_CSS = 'public-bundle.css';
+          const PUBLIC_BUNDLE_JS = 'public-bundle.js';
+          let checkTimestamp = false;
+
+          if (!fs.exists(bundleDir)) {
```

Comment:
You did not use `existsSync` or `await`. `exists` returns a promise, which is always truthy, so the mkdirp call never happens and the flag is never set.